python/nav/ipdevpoll/shadows/__init__.py

```diff
@@ -260,6 +264,51 @@ def _fix_binary_garbage(self):
                 setattr(self, attr, repr(value))
         self.clear_cached_objects()
 
+    def _get_new_device_event(self):
+        if not self.is_new:
+            return
+        device = self.get_existing_model()
+        if not device.serial:
+            return
+        # Check for chassis before module
+        # A device can be a module and a chassis, but chassis is prioritized
```

Comment:
Actually, I think you should team up with @johannaengland to build some common code here:

There is already a need to build a description of a `Device` object for human consumption, by making these exact lookups, and I believe @johannaengland is working on this for one of her related PRs.

I think this piece of lookup code you have here could be extracted into a new method on the `Device` model class.  E.g. `get_related_object()`. Johanna's new method that builds the description string could use the same method, and you could use it here to figure out what type of NAV object this device represents.

Reply:
If alerts should not be created when a Device is made, but rather when any Module/Fan/Psu etc is made, then I should probably move the logic to those respective shadow classes anyways. If i cant assume that the device is also new, then i'd have to go through every Device and check if the models that refer to it are new, which seems unnecessary if I can just do that directly in e.g. the Module shadow class

Reply:
I added a function `get_detailed_description` in #2413 in commit [4407b31](https://github.com/Uninett/nav/pull/2413/commits/4407b3154d01d530c5179fe9be18a82ec8e7f2bd)

Reply:
Should I now also add a function `get_related_object` that returns the related Module/Fan/PowerSupply/NetboxEntity if it exists?

Reply:
> If alerts should not be created when a Device is made, but rather when any Module/Fan/Psu etc is made, then I should probably move the logic to those respective shadow classes anyways. If i cant assume that the device is also new, then i'd have to go through every Device and check if the models that refer to it are new, which seems unnecessary if I can just do that directly in e.g. the Module shadow class

Agreed, @stveit! A Device record is only ever created as a result of a new NetboxEntity/Module/FanOrPSU record being created - but creating one of those don't necessarily lead to the creation of a Device record: An old one could be re-used. So the point is definitely to post events as the result of either adding a new NetboxEntity/Module/FanOrPSU or, potentially, altering an existing one to point to a different Device entry.

Reply:
> Should I now also add a function `get_related_object` that returns the related Module/Fan/PowerSupply/NetboxEntity if it exists?

Please do, @johannaengland , - but I think it would be cleanest if you added both functions as a separate PR :)